example/src/App.tsx

```diff
@@ -182,18 +188,32 @@ const ResultItem = React.forwardRef(
     {
       action,
       active,
+      currentRootActiveId,
     }: {
-      action: Action;
+      action: ActionImpl;
       active: boolean;
+      currentRootActiveId: ActionId;
     },
     ref: React.Ref<HTMLDivElement>
   ) => {
+    const ancestors = React.useMemo(() => {
```

Comment:
Could `ancestors` not be provided by `onRender` ?

Reply:
We can but we will get this behavior, notice the slight bit of ancestor flash when we hit backspace and go to the parent action:

https://user-images.githubusercontent.com/12195101/140166084-b80af246-4e7b-4d4e-9293-d23a27685799.mp4

Since we throttle the matching, the actions displayed and the `currentRootActionId` will not be synced perfectly. The current solution is to return the `currentRootActionId` alongside the matches so we ensure they're synced:

```
const { results, rootActionId } = useDeepMatches();
```

I may be missing something totally here bc I really don't like returning that `rootActionId` as part of `useDeepMatches` lol